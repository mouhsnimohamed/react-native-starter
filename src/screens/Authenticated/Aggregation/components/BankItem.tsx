import React from 'react';
import { Card, PrimaryText } from 'components';
import styled from 'styled-components/native';
import NavigateToCompte from 'assets/svg/navigateToCompte';
import { View } from 'react-native';
import { BankType } from 'modules/aggregation/aggregation.types';
import { useDispatch } from 'react-redux';
import { consentBankAction } from 'modules/aggregation/aggregation.actions';
export interface BankItem {
  item: BankType;
}
const BankItem = ({ item }: BankItem) => {
  const dispatch = useDispatch();
  const consentBank = () => dispatch(consentBankAction(item.id));
  return (
    <Card
      mTop={12}
      pTop={10}
      pBottom={1}
      mBottom={10}
      style={{ minHeight: 80 }}>
      <CompanyView onPress={consentBank}>
        <IconImg
          source={{
            uri: item.icon,
          }}
        />
        <PrimaryText
          style={{ paddingTop: 16, paddingLeft: 5 }}
          isPrimary
          semiBold
          align="left">
          {item.name}
        </PrimaryText>
        <View style={{ paddingTop: 12, paddingRight: 10 }}>
          <NavigateToCompte />
        </View>
      </CompanyView>
    </Card>
  );
};
export default BankItem;

const CompanyView = styled.TouchableOpacity`
  flex: 1;
  flex-direction: row;
  margin-bottom: 8px;
`;
const IconImg = styled.Image`
  width: 30px;
  height: 30px;
  border-radius: 60px;
  text-align: center;
  line-height: 80px;
  padding: 30px;
  margin-right: 8px;
`;
