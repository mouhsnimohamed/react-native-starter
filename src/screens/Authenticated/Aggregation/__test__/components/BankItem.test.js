import React from 'react';
import { renderWithThemeProvider, cleanup } from 'utils/test/render.utils';
import BankItem from '../../components/BankItem';
import * as reactRedux from 'react-redux';

afterEach(cleanup);

describe('<BankItem />', () => {
  it('render BankItem correctly', () => {
    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);

    const rendered = renderWithThemeProvider(
      <BankItem
        item={[
          { bankName: 'Credit agricole', iconName: 'iconName' },
          { bankName: 'Revolut', iconName: 'iconName' },
          { bankName: 'Bank of africa', iconName: 'iconName' },
        ]}
      />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
