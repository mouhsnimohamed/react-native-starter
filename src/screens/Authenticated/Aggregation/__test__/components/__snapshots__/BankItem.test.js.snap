// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankItem /> render BankItem correctly 1`] = `
<View
  mBottom={10}
  mTop={12}
  pBottom={1}
  pLeft={13}
  pRight={13}
  pTop={10}
  radius={11}
  style={
    Array [
      Object {
        "backgroundColor": "#F8F8F9",
        "borderColor": "#BFCEDE",
        "borderRadius": 11,
        "borderStyle": "solid",
        "borderWidth": 1,
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 12,
        "paddingBottom": 1,
        "paddingLeft": 13,
        "paddingRight": 13,
        "paddingTop": 10,
        "width": "100%",
      },
      Object {
        "minHeight": 80,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 8,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 60,
            "height": 30,
            "lineHeight": 80,
            "marginRight": 8,
            "paddingBottom": 30,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 30,
            "textAlign": "center",
            "width": 30,
          },
        ]
      }
    />
    <Text
      align="left"
      isPrimary={true}
      semiBold={true}
      style={
        Array [
          Object {
            "color": "#003D7C",
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
            "width": "100%",
          },
          Object {
            "paddingLeft": 5,
            "paddingTop": 16,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "paddingRight": 10,
          "paddingTop": 12,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="31"
        bbWidth="31"
        fill="none"
        focusable={false}
        height="31"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 31,
              "width": 31,
            },
          ]
        }
        vbHeight={26}
        vbWidth={26}
        width="31"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M10.2305 18.4984C10.5367 18.8047 11.0305 18.8047 11.3367 18.4984L16.5305 13.3047C16.7742 13.0609 16.7742 12.6672 16.5305 12.4234L11.3367 7.22969C11.0305 6.92344 10.5367 6.92344 10.2305 7.22969C9.92422 7.53594 9.92422 8.02969 10.2305 8.33594L14.7555 12.8672L10.2242 17.3984C9.92422 17.6984 9.92422 18.1984 10.2305 18.4984Z"
            fill={4278205820}
            propList={
              Array [
                "fill",
              ]
            }
          />
          <RNSVGCircle
            cx="13"
            cy="13"
            propList={
              Array [
                "stroke",
              ]
            }
            r="12.5"
            stroke={4278205820}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
